app/list_cmd.go

```diff
@@ -114,34 +115,55 @@ func listPackagesInCLI(pkgs manifest.Packages, option *listPackageOption) {
 		w = 80
 	}
 
+	if option.Prefix != "" {
+		for i, pkg := range pkgs {
+			if strings.HasPrefix(pkg.Reference.Name, option.Prefix) {
+				pkgs = append(pkgs[:i], pkgs[i+1:]...)
+			}
+		}
+	}
+	// for each name, check if it's a prefix match
+	pkgMiss := make([]string, 0, len(names))
 	for _, name := range names {
```

Comment:
I think there's a simpler way to do all this. If you replace [this sort.Strings()](https://github.com/cashapp/hermit/blob/master/app/list_cmd.go#L80) call with a `sort.Slice()` call, you should be able to construct a "less" function that correctly sorts by prefix, then alphabetically. That should do the right thing, I believe.

Don't worry too much about tests. The reason they're not working is because this part of the CLI bypasses the UI buffer and writes directly to stdout. Not something you should tackle in this PR.

Reply:
Actually scratch that, the sort will have to be in this function, so you'll need to pass the constraint down as an (optional) search option.

Reply:
Which you have already!

Reply:
cool -- we need to add 'state' to that, the `Less(i, j int)` doesn't have an option to pass in the prefix to match on, so I created a new struct that stores the list to sort plus the prefix string (so the compare function does work)

Reply:
An intermediate structure isn't  necessary, a closure will capture the prefix.

Reply:
nice, that's much simpler - updated with that approach